src/mpid/ch4/src/ch4r_callbacks.c

```diff
@@ -526,11 +545,18 @@ int MPIDIG_send_pipeline_rts_target_msg_cb(int handler_id, void *am_hdr, void *d
          * may remain valid by the time we send ack (using the comm).
          */
         MPIDIG_REQUEST(rreq, req->status) |= MPIDIG_REQ_PIPELINE_RTS;
-        MPIDIG_REQUEST(rreq, req->rreq.peer_req_ptr) = rts_hdr->sreq_ptr;
+        MPIDIG_REQUEST(rreq, req->peer_req_ptr) = rts_hdr->sreq_ptr;
         MPIDIG_REQUEST(rreq, rank) = hdr->src_rank;
         MPIDIG_REQUEST(rreq, tag) = hdr->tag;
         MPIDIG_REQUEST(rreq, context_id) = hdr->context_id;
         MPIDIG_REQUEST(rreq, req->status) |= MPIDIG_REQ_IN_PROGRESS;
+        MPIDIG_REQUEST(rreq, req->target_cmpl_cb) = recv_target_cmpl_cb;
```

Comment:
I might need to revise this again because the handling for RMA callback would happen at a different place.